ros2_batch_job/packaging.py

```diff
@@ -159,7 +159,14 @@ def build_and_test_and_package(args, job):
     # create an archive
     folder_name = 'ros2-' + args.os
     if args.os == 'linux' or args.os == 'osx':
-        archive_path = 'ros2-package-%s-%s.tar.bz2' % (args.os, platform.machine())
+        machine = platform.machine()
+        # Detect when we're using a 32-bit interpreter on a 64-bit platform
+        if sys.maxsize <= 2**32:
+            if machine == 'aarch64':
```

Comment:
I wonder if we're ever going to add non-aarch64 build hosts, and if so this might be a subtle problem to find then. Perhaps `if machine == 'aarch64' or 'arm' in machine:` (looking at values in https://stackoverflow.com/questions/45125516/possible-values-for-uname-m)

Reply:
I found another way. What we're really interested in is the version of the OS files, not the kernel or the hardware. The best approximation I can find is to look at `sys.implementation`'s multiarch value. This returns the typical values for `x86_64`, `i386`, and `aarch64`, and gives us `arm` for 32-bit arm. Not the `armhf` we were looking for, but probably less hacky.

Thoughts? Maybe I should special case `arm` to `armhf` for clarity? We have no precedent for 32-bit arm archives.

Reply:
This approach is much nicer than the first pass. I think we still want to output the archive as `armhf`, so the special case seems reasonable to me.

Reply:
Updated and triggered new CI to verify.